src/lcov.ts

```diff
@@ -119,20 +114,29 @@ export function getProjectLineCoverageBefore(
 export function getTotalPercentage(
   projects: CoveredProject[]
 ): LineCoverage | undefined {
-  const coverages = projects
-    .map(p => p.coverage)
-    .flat()
-    .filter(c => c !== undefined && c !== null) as parse.LcovFile[]
-
-  return getLineCoverage(coverages)
+  return getLineCoverage(getProjectCoverage(projects, 'coverage'))
 }
 
 export function getTotalPercentageBefore(
   projects: CoveredProject[]
-): LineCoverage | undefined {
-  const coverages = projects
-    .map(p => p.coverageBefore)
-    .flat()
-    .filter(c => c !== undefined && c !== null) as parse.LcovFile[]
-  return getLineCoverage(coverages)
+): LineCoverage {
+  return getLineCoverage(getProjectCoverage(projects, 'coverageBefore'))
+}
+
+function getProjectCoverage(
```

Comment:
I would suggest renaming this to something like `getSectionSummaries`. I think the word "coverage" has a confusing meaning in this file already, and this adds to that a little bit :)

Reply:
I've renamed the `getProjectCoverage` function to `getSectionSummaries` as you suggested and also replaced the inner `for..of` loop with a `<Array>.concat` call (since it's technically the same, but requires less code in this file).